packages/block-library/src/image/save.js

```diff
@@ -42,6 +42,7 @@ export default function save( { attributes } ) {
 			width={ width }
 			height={ height }
 			title={ title }
+			data-id={ id ? id : null }
```

Comment:
This means, we'd need a deprecation for the image block, given that the id is already in the "className", I'm not sure this attribute is necessary at all personally.

Reply:
@youknowriad it was added to provide some backwards compat of the new gallery format for themes and plugins that rely on `data-id` in the gallery image tag - https://github.com/WordPress/gutenberg/pull/30710. Happy to take guidance on whether we add a deprecation or remove this and communicate to plugin/theme authors that they will have to add a fallback to the classname if they don't already do so.

Reply:
It does seem to me like plugins won't probably work 1-1 anyway and will require some changes to adapt to the new markup... In that case, they can probably be updated to rely on the class right?

Reply:
Previously the `data-id` was only on gallery images so depending on how the plugin targeted them, they might have had to tighten up their selectors as well. I think that further supports removing the data-id and avoiding the deprecation. Especially given how many image blocks could be on a page.

Reply:
Thanks for the input, have reverted this for now.